server/server.py

```diff
@@ -1,113 +1,125 @@
+            'datefmt': '%Y-%m-%d %H:%M:%S',
+        },
+        "colored": {
+            'format': '{Time: %(asctime)s, '
+                      'Level: [%(levelname)s], ' +
+                      'module: %(module)s, '
+                      'function: %(funcName)s():%(lineno)s, ' +
+                      'message: %(message)s}',
+            "()": structlog.stdlib.ProcessorFormatter,
+            "processor": structlog.dev.ConsoleRenderer(colors=True),
+            'datefmt': '%Y-%m-%d %H:%M:%S',
+        }},
+    'handlers': {
+        'wsgi': {
+            'class': 'logging.StreamHandler',
+            'stream': 'ext://flask.logging.wsgi_errors_stream',
+            'formatter': 'colored'
+        }
+    },
+    'root': {
+        'level': 'INFO',
+        'propogate': True,
+        'handlers': ['wsgi']
+    }
+})
+
+app = Flask(__name__)
+# HTTP security header middleware for Flask
+talisman = Talisman(app)
+talisman.force_https = False
+config = cast(Dict[str, Any], toml.load('config.toml'))
+credentials = Credentials(config['creds_path'])
+core = make_core(config, credentials)
+webhook_handler = make_webhook_handler(config, credentials)
+if not config['testing']:
+    slack_signing_secret = credentials.slack_signing_secret
+else:
+    slack_signing_secret = ""
+slack_events_adapter = SlackEventAdapter(slack_signing_secret,
+                                         "/slack/events", app)
+
+
+@app.route('/')
+def check():
+    """Display a Rocket status image."""
+    logging.info('Served check()')
+    return "🚀"
+
+
+@app.route('/slack/commands', methods=['POST'])
+def handle_commands():
+    """Handle rocket slash commands."""
+    txt = request.form['text']
+    uid = request.form['user_id']
+    return core.handle_app_command(txt, uid)
+
+
+@app.route('/webhook/organization', methods=['POST'])
+def handle_organization_webhook():
+    """Handle GitHub organization webhooks."""
+    logging.info("organization webhook triggered")
+    xhub_signature = request.headers.get('X-Hub-Signature')
+    request_data = request.get_data()
+    request_json = request.get_json()
+    logging.debug(f"organization payload: {str(request_json)}")
+    if webhook_handler.verify_hash(request_data, xhub_signature) is True:
+        return webhook_handler.handle_organization_event(request_json)
+    else:
+        return "Hashed signature is not valid", 403
+
+
+@app.route('/webhook/team', methods=['POST'])
+def handle_team_webhook():
+    """Handle GitHub team webhooks."""
+    logging.info("team webhook triggered")
+    xhub_signature = request.headers.get('X-Hub-Signature')
+    request_data = request.get_data()
+    request_json = request.get_json()
+    logging.debug(f"team payload: {str(request.get_json())}")
+    if webhook_handler.verify_hash(request_data, xhub_signature) is True:
+        msg = webhook_handler.handle_team_event(request.get_json())
+    else:
+        msg = "Hashed signature is not valid", 403

```

Comment:
This hash verification looks like it could be abstracted out into a separate function.

Reply:
Not sure what exactly this means but see what I did I guess?